ddl/constraint.go

```diff
@@ -86,15 +87,25 @@ func (w *worker) onAddCheckConstraint(d *ddlCtx, t *meta.Meta, job *model.Job) (
 	originalState := constraintInfoInMeta.State
 	switch constraintInfoInMeta.State {
 	case model.StateNone:
-		// none -> write only
 		job.SchemaState = model.StateWriteOnly
 		constraintInfoInMeta.State = model.StateWriteOnly
 		ver, err = updateVersionAndTableInfoWithCheck(d, t, job, tblInfo, originalState != constraintInfoInMeta.State)
 	case model.StateWriteOnly:
-		// write only -> public
-		err = w.verifyRemainRecordsForCheckConstraint(dbInfo, tblInfo, constraintInfoInMeta, job)
-		if err != nil {
-			return ver, errors.Trace(err)
+		job.SchemaState = model.StateWriteReorganization
+		constraintInfoInMeta.State = model.StateWriteReorganization
+		ver, err = updateVersionAndTableInfoWithCheck(d, t, job, tblInfo, originalState != constraintInfoInMeta.State)
+	case model.StateWriteReorganization:
+		skipCheck := false
+		failpoint.Inject("mockVerifyRemainDataSuccess", func(val failpoint.Value) {
```

Comment:
Maybe we can put it to `verifyRemainRecordsForCheckConstraint`, and if `mockVerifyRemainDataSuccess` is ture, we can return this function. Then we needn't add `skipCheck`.